### PR TITLE
Add Ruff

### PR DESCRIPTION
**Pull request by shikanime**
Ruff is a Python linter that is starting to get a lot of attention lately.

### CODE REVIEW

File: modules/hooks.nix [hunk 1]
```diff
@@ -359,6 +359,13 @@ in
           entry = "${pkgs.python3Packages.black}/bin/black";
           types = [ "file" "python" ];
         };
+      ruff =
+        {
+          name = "ruff";
+          description = " An extremely fast Python linter, written in Rust.";
+          entry = "${pkgs.python3Packages.black}/bin/ruff";
+          types = [ "file" "python" ];
```
Comment (loicreynier) [inline modules/hooks.nix:367]:
```suggestion
          types = [ "python" ];
```

The type `file` is not necessary when using the `python` type.

File: modules/hooks.nix [hunk 2]
```diff
@@ -359,6 +359,13 @@ in
           entry = "${pkgs.python3Packages.black}/bin/black";
           types = [ "file" "python" ];
         };
+      ruff =
+        {
+          name = "ruff";
+          description = " An extremely fast Python linter, written in Rust.";
+          entry = "${pkgs.python3Packages.black}/bin/ruff";
```
Comment (loicreynier) [inline modules/hooks.nix:366]:
```suggestion
          entry = "${pkgs.ruff}/bin/ruff";
```

`ruff` is not a Python package.

**Review by loicreynier — CHANGES_REQUESTED** [review]

File: nix/tools.nix [hunk 3]
```diff
@@ -53,7 +53,7 @@ let
   };
 in
 {
-  inherit actionlint alejandra cabal-fmt cabal2nix cargo clang-tools clippy deadnix dhall editorconfig-checker hadolint hindent hlint hpack html-tidy nix-linter nixfmt nixpkgs-fmt ormolu rustfmt shellcheck shfmt statix stylish-haskell stylua typos go mdsh revive go-tools;
+  inherit actionlint alejandra cabal-fmt cabal2nix cargo clang-tools clippy deadnix dhall editorconfig-checker hadolint hindent hlint hpack html-tidy nix-linter nixfmt nixpkgs-fmt ormolu rustfmt shellcheck shfmt statix stylish-haskell stylua typos go mdsh revive go-tools ruff;
```
Comment (loicreynier) [inline nix/tools.nix:56]: `ruff` should also be added to the arguments line 32.

**Review by loicreynier — CHANGES_REQUESTED** [review]

**Comment by shikanime** [comment]
Oh, at the moment, it seems that Ruff is only available on Nixpkgs version 22.11.

**Comment by domenkozar** [comment]
I'll try to bump checks to 22.11

**Comment by domenkozar** [comment]
If you rebase it should work now

**Comment by dhess** [comment]
FYI, I believe this commit broke the flake. We now get:

```
error: Function called without required argument "ruff" at /nix/store/lvwknwz9wfx66qyxdcak5p6w2wfxlzpd-source/nix/tools.nix:32, did you mean "duff", "guff" or "rufo"?
```

**Comment by shikanime** [comment]
Please correct me if I'm wrong, as the package is only available in 22.11 release, should I have added a default to null ?

**Comment by dhess** [comment]
@shikanime Possibly. This happens when I use `pre-commit-hooks.nix` with `haskell.nix`, which is, I believe, using a pre-22.11 nixpkgs-unstable pin. There's a PR to bump it: https://github.com/input-output-hk/haskell.nix/pull/1815

**Comment by shikanime** [comment]
Thanks for the extra details, I didn't realize the follows mechanism worked like that after reading the documentation.
I haven't had much experience with Flake dependencies system.
Sorry for the inconveniences (⁠｡⁠ŏ⁠﹏⁠ŏ⁠)

**Comment by dhess** [comment]
No worries at all, thanks for the contribution.

**Pull request merged**